lib/clients/authenticated.js

```diff
@@ -163,21 +185,14 @@ class AuthenticatedClient extends PublicClient {
             callback(undefined, response, totalDeletedOrders);
           }
           return totalDeletedOrders;
-        })
-        .catch(err => {
-          if (callback) {
-            callback(err);
-          }
-          throw err;
         });
     }.call(this);
 
     if (callback) {
-      p.catch(() => {});
```

Comment:
Great comment. I'll wrap this in a `setImmediate`. In general I would opt for always returning a promise and assigning listeners to it in case there is a callback.